src/plugins/es_ui_shared/public/components/view_api_request_flyout/view_api_request_flyout.tsx

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+// We want to allow both right-clicking to open in a new tab and clicking through
+// the "Open in Console" link. We could use `RedirectAppLinks` at the top level
+// but that inserts a div which messes up the layout of the flyout.
+/* eslint-disable @elastic/eui/href-or-on-click */
+
+import React, { useCallback } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { compressToEncodedURIComponent } from 'lz-string';
+
+import {
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiButtonEmpty,
+  EuiText,
+  EuiSpacer,
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiCopy,
+} from '@elastic/eui';
+import { ApplicationStart } from 'src/core/public';
+import type { UrlService } from 'src/plugins/share/common/url_service';
+
+interface ViewApiRequestFlyoutProps {
+  title: string;
+  description: string;
+  request: string;
+  closeFlyout: () => void;
+  navigateToUrl?: ApplicationStart['navigateToUrl'];
+  urlService?: UrlService;
+  canShowDevtools?: boolean;
+}
+
+export const ViewApiRequestFlyout: React.FunctionComponent<ViewApiRequestFlyoutProps> = ({
+  title,
+  description,
+  request,
+  closeFlyout,
+  navigateToUrl,
+  urlService,
+  canShowDevtools,
+}) => {
+  const getUrlParams = undefined;
+  const devToolsDataUri = compressToEncodedURIComponent(request);
+
+  // Generate a console preview link if we have a valid locator
+  const consolePreviewLink = urlService?.locators.get('CONSOLE_APP_LOCATOR')?.useUrl(
+    {
+      loadFrom: `data:text/plain,${devToolsDataUri}`,
+    },
+    getUrlParams,
+    [request]
+  );
+
+  const consolePreviewClick = useCallback(
+    () => consolePreviewLink && navigateToUrl && navigateToUrl(consolePreviewLink),
+    [consolePreviewLink, navigateToUrl]
+  );
+
+  // Check if both the Dev Tools UI and the Console UI are enabled.
+  const shouldShowDevToolsLink = canShowDevtools && consolePreviewLink !== undefined;
+
+  return (
+    <EuiFlyout maxWidth={480} onClose={closeFlyout} data-test-subj="apiRequestFlyout">
```

Comment:
Is there a specific reason to have a hard coded `maxWidth`? While testing locally, I noticed that the flyout could be larger to better display the pipeline code. See screenshot below, I use a large monitor though.
I would suggest allow passing `maxWidth` as a prop and use some larger value in Ingest Pipelines. 
<img width="2458" alt="Screenshot 2022-03-16 at 16 40 10" src="https://user-images.githubusercontent.com/6585477/158629475-f5f18a58-7dff-46ba-a6e4-a10b5ce4efaa.png">

Reply:
Good point! I'll add a new prop to allow to configure the flyout props to consumers